pkg/api/types.go

```diff
@@ -200,6 +200,8 @@ type VolumeSource struct {
 	ISCSI *ISCSIVolumeSource `json:"iscsi"`
 	// Glusterfs represents a Glusterfs mount on the host that shares a pod's lifetime
 	Glusterfs *GlusterfsVolumeSource `json:"glusterfs"`
+	// CinderPersistentDisk represents a cinder volume attached and mounted on kubelets host machine
+	CinderPersistentDisk *CinderPersistentDiskVolumeSource `json:"cinderPersistentDisk"`
```

Comment:
Provisioning the volumes themselves is an orthogonal task to this.  So long as the volumes exist in the infrastructure (however they were provisioned), they can be PersistentVolumes.  You just need to implement the PersistentVolumePlugin interface and make sure the pointer is in both structs.